libsel4vm/src/arch/arm/vgic/vgic.c

```diff
@@ -194,24 +194,42 @@ struct irq_queue {
     size_t tail;
 };
 
+typedef struct vgic_core {
+    // Mirrors the vcpu list registers
+    struct virq_handle *lr_shadow[NUM_LIST_REGS];
+    // IRQs that would not fit in the vcpu list registers
+    struct irq_queue irq_queue;
+    // Complete set of virtual irqs
+    struct virq_handle *sgi_ppi_irq[NUM_SGI_VIRQS + NUM_PPI_VIRQS];
+} vgic_core_t;
```

Comment:
I wonder, would `vgic_vcpu_t` be a better name?

Reply:
Good idea.